src/api/ppx_deriving.cppo.ml

```diff
@@ -876,6 +816,6 @@ let hash_variant s =
 (* This is only used when ppx_deriving is linked as part of an ocaml-migrate-parsetre
    driver. *)
 let () =
-  Migrate_parsetree.Driver.register ~name:"ppx_deriving"
-    (module Migrate_parsetree.OCaml_current)
-    (fun _ _ -> mapper)
+  Ppxlib.Driver.register_transformation "ppx_deriving"
+    ~preprocess_impl:map_structure
```

Comment:
`preprocess_intf` and `preprocess_impl` are special fields used to define a transformation that must run before anything else and as it turns out you can only use one such ppx at any given time, this is why it's incompatible with `ppx_optcomp`. I'm guessing this is required because ppxlib otherwise throws warnings about uninterpreted deriving plugins right?